libs/README.md

```diff
@@ -0,0 +1,40 @@
+# Libs
+
+The `libs` directory is home to any libraries shared across studio applications.
+
+## Creating a new library
+
+A library has 3 requirements
+
+1. It must include a `tsconfig.json` with at least the following contents
+
+```json
+{
+  "extends": "../../config/tsconfig.lib.json",
+  "include": ["src/**/*.tsx", "src/**/*.ts", "types/*.ts"],
+  "compilerOptions": {
+    "rootDir": "src",
+    "outDir": ".ts-out"
+  }
+}
+```
+
+2. It must contain a `src` directory where all the code lives
+3. It must contain and `index.ts` which exports everything that is usable outside the library
+
+## Using a library
+
+If you've created a library and want to use it somewhere else in studio (either in an app or a library), there are two things you need to know
+
+1. To import the library from somewhere else use `@artsy-studio/<library-name>`
+2. You _must_ add the `<library-name>` as `references` in the `tsconfig.json` of the place it's used.
+
+For example, if you created a library call `widgets` and wanted to use it in the `studio` app, you would add the following to the `studio` app's `tsconfig.json`:
+
+```json
+{
+  "references": [{ "path": "../../widgets" }]
+}
+```
+
+It's a bit cumbersome for now, but it's on the roadmap to automate away this requirement.
```

Comment:
Does all of this add more complexity than conventional workflows? With workspaces and other such models this kind of paperwork isn't necessary, and automating it away means creating custom tooling for something that isn't necesarily problematic.

Reply:
I think the added complexity of lerna + workspaces is way above this. You don't need to learn another tool. You don't need tk think about bootstrap or linking. These references *do* have to be added, but that's pretty easy to automate.

Reply:
Do folks think about those things when using palette these days? The lerna integration is pretty seamless when working on the docs for example, and yarn workspaces is pretty much zero config.

Reply:
Thinking further, why do we need to imagine divisions between apps that are larger than routes / well-organized folder structures? Next is designed to handle large apps with bundle splitting and so on. It seems like a single package.json and a single tsconfig would be enough?

Reply:
For now, this is just Typescript and nothing else. It's simple and relatively simple to wrap your head around. If it turns out that it doesn't work out, it's a relatively low lift to add lerna + workspaces. The structure can stay the same, we'd just have to reconfigure how ts compiles to remove composites. If nothing else typescript project references and workspaces can live together. We'd _still_ have to manually (or with automation) configure the references. @orta [mentioned](https://artsyalumni.slack.com/archives/CJFV3E8QH/p1587396836055500?thread_ts=1587394998.052800&cid=CJFV3E8QH) there's talk of automatically configuring references in a workspaces/lerna monorepo. If that happens, it's easy enough to move to that. 

Given that we can add it without a significant refactor, I say we wait to make that decision. Let me play around with tweaking references. If it doesn't work out, it doesn't work out and we have other viable options.